c10/core/TensorOptions.h

```diff
@@ -738,9 +743,12 @@ inline Layout dispatchKeyToLayout(DispatchKey dispatch_key) {
     case DispatchKey::SparseHIP:
     case DispatchKey::SparseVE:
     case DispatchKey::SparseXPU:
+      return Layout::Sparse;
     case DispatchKey::SparseCsrCPU:
     case DispatchKey::SparseCsrCUDA:
-      return Layout::Sparse;
+      // warning: existence of Layout::SparseCsc makes the
+      // dispatchKeyToLayout function a multi-valued mapping
```

Comment:
The only call site for this function is [the legacy constructor](https://github.com/pytorch/pytorch/blob/a9b0a921d592b328e7e80a436ef065dadda5f01b/torch/csrc/autograd/python_legacy_variable.cpp). Since this function is currently wrong, but somehow not failing let's address it separately. I'd suggest we don't touch it in the context of this PR. cc @ezyang since you've been looking into them